### PR TITLE
Feature/histogram ci plot

### PR DESCRIPTION
**Pull request by zackt113**
Added the plot function. I only made a histogram with ci and observed mean, as when I went over the inference course we took in Block 2, they don't really use the curve and the histogram looks good for showing density and ci as well. Please let me know if there's anything you would like to change! 

Thanks,
Zack

### CODE REVIEW

File: src/straPy/data_vis.py [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+    if not isinstance(x_axis, str):
+        raise TypeError("The value of the argument 'x_axis' must be type of str.")
+        
+    if not isinstance(y_axis, str):
+        raise TypeError("The value of the argument 'y_axis' must be type of str.")
+        
+    plt.hist(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[1], density=False, bins=bin_size)
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["lower"], color='k', linestyle='--')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["sample_mean"], color='r', linestyle='-')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["upper"], color='k', linestyle='--')
```
Comment (gauthampughazhendhi) [inline src/straPy/data_vis.py]: @zackt113, just a small formatting change, the code lines according to pep8 standard should not be more than 79 characters.

File: src/straPy/data_vis.py [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
     -------
-    plot object
-        density plot with confidence interval
+    plot: histogram
+        histogram of bootstrapped distribution with confidence interval and observed mean
     
     Examples
     --------
-    >>> density_ci_plot([1, 1, 2, 3, 5, 10]，[1., 9.375]）
+    >>> histogram_ci_plot([1, 2, 3, 4, 5, 6, 7], 1000 ,n=100, ci_level=0.95, ci_random_seed=123)
```
Comment (gauthampughazhendhi) [inline src/straPy/data_vis.py:41]:
```suggestion
    >>> histogram_ci_plot([1, 2, 3, 4, 5, 6, 7], 1000, n=100, ci_level=0.95, ci_random_seed=123)
```

**Review by gauthampughazhendhi — CHANGES_REQUESTED** [review]
@zackt113, the work looks awesome. I have one suggestion if you have time, can we add a text component near the mean's vertical line that has mean (+/-) standard error in it. Ex: 2.63 (+/- 0.2)? Also, text components for confidence level lines, just to make the plot more attractive.

**Comment by zackt113** [comment]
Hi @gauthampughazhendhi, just fixed the function based on your suggestion please take a look. Thank you!

**Comment by voremargot** [comment]
Things look good to me! I would be happy if you merged!

**Review by gauthampughazhendhi — APPROVED** [review]
The changes look good to me @zackt113. Good work!

**Pull request merged**